examples/llm_compression/torch/qat_with_nls_downstream/main.py

```diff
@@ -0,0 +1,773 @@
+    if specific_rank_config is not None:
+        configure_lora_adapters(
+            get_layer_id_vs_lora_quantizers_map(model_to_eval),
+            specific_rank_config=specific_rank_config,
+        )
+    model_to_eval = nncf.strip(model_to_eval, do_copy=False, strip_format=StripFormat.DQ, example_input=example_input)
+    export_from_model(model_to_eval, ir_dir, device="cpu")
+    return OVModelForCausalLM.from_pretrained(
+        model_id=ir_dir,
+        trust_remote_code=True,
+        load_in_8bit=False,
+        compile=True,
+        ov_config={"KV_CACHE_PRECISION": "f16", "DYNAMIC_QUANTIZATION_GROUP_SIZE": "0"},
+    )
+
+
+def get_argument_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(add_help=True)
+
+    # Model params
+    parser.add_argument(
+        "--pretrained",
+        type=str,
+        default="Qwen/Qwen2.5-3B-Instruct",
+        help="The model id or path of a pretrained HF model configuration.",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=Path,
+        default="output",
+        help="Path to the directory for storing logs, tuning checkpoint, compressed model, validation references.",
+    )
+    parser.add_argument(
+        "--resume",
+        action="store_true",
+        help="Whether to start from previously saved checkpoint. If not specified or checkpoint does not exist, "
+        "start from scratch by post-training weight compression initialization.",
+    )
+    parser.add_argument(
+        "--do_train",
+        action="store_true",
+    )
+
+    # Downstream task
+    parser.add_argument(
+        "--task",
+        type=str,
+        choices=[
+            "openbookqa",
+            "winogrande",
+            "arc_challenge",
+            "arc_easy",
+            "gsm8k",
+            "hellaswag",
+        ],
+        default="openbookqa",
+        help="Evaluation task",
+    )
+    parser.add_argument(
+        "--lm_eval_metric",
+        type=str,
+        default="acc_norm,none",
+        help="The metrics of the lm-eval task. Different tasks have different metrics.",
+    )
+
+    # Training params
+    parser.add_argument(
+        "--lr",
+        type=float,
+        default=1e-4,
+        help="Learning rate for fine-tuning. "
+        "For larger models (over 2 billion parameters), a learning rate of 5e-4 is recommended.",
+    )
+    parser.add_argument("--epochs", type=int, default=3, help="Number of epochs.")
+    parser.add_argument("--batch_size", type=int, default=16, help="Size of training batch.")
+    parser.add_argument(
+        "--microbatch_size",
+        type=int,
+        default=16,
+        help="Size of each training microbatch. Gradients will be accumulated until the batch size is reached.",
+    )
+    parser.add_argument("--eval_batch_size", type=int, default=64, help="Size of batch for evaluation.")
+
+    # Neural Low-rank Adapter Search (NLS) params
+    parser.add_argument(
+        "--lora_rank_space",
```

Comment:
make it `required` or specify some defaults.
otherwise, it fails by default with not clear error.
![image](https://github.com/user-attachments/assets/7e8d6267-3e28-4eb1-baf8-00e993402069)

Reply:
@ljaljushkin Fixed with defaults. Thanks!